### PR TITLE
GITHUB-2207: Allow dependency injector factory to be configured via args (#2207)

### LINKED ISSUE #2207: Dependency injector factory can't be configured via Maven surefire plugin
### TestNG Version
org.testng:testng:7.1.0

### Expected behavior
If the custom implementation class of `org.testng.IInjectorFactory` is passed via configuration properties of Maven surefire plugin it should be applied to executable tests.

### Actual behavior
Specified configuration option is ignored.

### Is the issue reproductible on runner?

- [ ] Shell
- [x] Maven
- [ ] Gradle
- [ ] Ant
- [ ] Eclipse
- [ ] IntelliJ
- [ ] NetBeans

### Test case sample
Sample of Surefire plugin definition:
 ```  
          <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-surefire-plugin</artifactId>
                <version>2.22.2</version>
                <configuration>
                    <properties>
                        <property>
                            <name>dependencyinjectorfactory</name>
                            <value>customimplementationclass</value>
                        </property>
                    </properties>
                </configuration>
            </plugin>
```

### PR DESCRIPTION
**Pull request by dmikhievich**
Closes #2207

Fixes #2207.

### Did you remember to?

- [x] Add test case(s)
- [x] Update `CHANGES.txt`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

### CODE REVIEW

File: src/test/java/org/testng/TestNGConfigureTest.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package org.testng;
```
Comment (juherr) [inline src/test/java/org/testng/TestNGConfigureTest.java]: Use `test.commandline` or `test.cli` package instead

File: src/test/resources/testng.xml [hunk 2]
```diff
@@ -833,6 +833,7 @@
       <class name="test.configuration.github1625.TestRunnerIssue1625"/>
       <class name="test.configuration.issue1753.IssueTest"/>
       <class name="test.configuration.BeforeClassTest"/>
+      <class name="org.testng.TestNGConfigureTest"/>
```
Comment (juherr) [inline src/test/resources/testng.xml]: not the best section: here, configuration means configuration method

Maybe you can add a new cli/commandline section

**Comment by krmahadevan** [comment]
@juherr - Do you think we should go ahead and get the changes done ourselves and proceed with getting this changeset merged ? I can do that.  Please let me know.

**Comment by juherr** [comment]
@krmahadevan Yes, feel free to update: the change is cool and it will be sad block it due to some style issues.

**Comment by krmahadevan** [comment]
This has been merged as part of https://github.com/cbeust/testng/pull/2216

**Pull request closed without merging**